services/wallet/api.go

```diff
@@ -589,7 +590,58 @@ func (api *API) AddressExists(ctx context.Context, address types.Address) (bool,
 	return api.s.accountsDB.AddressExists(address)
 }
 
-// Returns details for the passed address (response doesn't include derivation path)
+// GetAddressDetailsAcrossAllChainsWithTimeout returns details for passed address (response doesn't include derivation path)
+// across all chains, with a timeout (passed in Milliseconds) for the call to complete
+func (api *API) GetAddressDetailsAcrossAllChainsWithTimeout(ctx context.Context, address string, timeoutInMilliseconds int64) (*DerivedAddress, error) {
```

Comment:
We don't really need 2 endpoints for this. 
Basically `timeout` is an optional argument. Just keep a special value (e.g. `0`) meaning "no timeout".

Reply:
At the same time, we encourage to have `json` strings in API endpoints. This allows to extend the API in future without breaking backwards compatibility. 

And it would help here as well, you could make `timeoutInMilliseconds` of `*int64` pointer type. So when it's `nil` (not provided) would mean "no timeout".

Take a look at `protocol/requests` directory for examples.

Reply:
@igor-sirotin nice suggestion, updated, thanks.

Reply:
Thank you!